### PR TITLE
Improve support for environment autodiscovery by removing explicit setting of DOCKER_HOST by default with Agent 7.27+

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

- Deploy Agent >= 7.27 with Operator, `DOCKER_HOST` should not be set and `/var/run` should be mounted to `/host/var/run`
- Deploy Agent < 7.27 with Operator, `DOCKER_HOST` should be set and `/var/run` should be mounted to `/host/var/run`

### CODE REVIEW

File: api/v1alpha1/datadogagent_default.go [hunk 1]
```diff
@@ -532,52 +533,51 @@ func DefaultDatadogAgentSpecAgentImage(image *ImageConfig) *ImageConfig {
...
+		agent.Config.CriSocket = &CRISocketConfig{}
 	}
 
-	if config.CriSocket == nil {
-		config.CriSocket = &CRISocketConfig{
-			DockerSocketPath: NewStringPointer(defaultDockerSocketPath),
+	// Don't default Docker/CRI paths with Agent >= 7.27.0
+	// Let Env AD do the work for us
+	agentTag := utils.GetTagFromImageName(agent.Image.Name)
+	if !(agentTag == "latest" || utils.IsAboveMinVersion(agentTag, "7.27.0")) {
```
Comment (ahmed-mez) [inline api/v1alpha1/datadogagent_default.go:560]: does this support agent 6 and jmx images?

File: pkg/utils/version.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2021 Datadog, Inc.
+
+package utils
+
+import "github.com/Masterminds/semver/v3"
+
+// IsAboveMinVersion uses semver to check if `version` is >= minVersion
+func IsAboveMinVersion(version, minVersion string) bool {
```
Comment (ahmed-mez) [inline pkg/utils/version.go:11]: mind adding a unit test? 🙇

File: api/v1alpha1/datadogagent_default.go (hunk 1 above)
Comment (vboulineau) [reply api/v1alpha1/datadogagent_default.go]: Good catch, changed.

File: pkg/utils/version.go (hunk 2 above)
Comment (vboulineau) [reply pkg/utils/version.go:11]: Done

**Review by ahmed-mez — APPROVED** [review]

**Pull request merged**